driver-kncasic.c

```diff
@@ -72,6 +72,7 @@ struct knc_core_state {
 		int slot;
 		uint32_t nonce;
 	} last_nonce;
+	uint32_t last_nonce_verified;
```

Comment:
The majority of those duplicates is caused by communication errors due to external noise and crosstalks in the ribbon cable. So, basically they are expected; that's why I want to filter them out.
As to how frequent they are: in some setups I have seen up to 2% of duplicates, depending on level of external noise.
That exposure to the communication errors is something that was expected. We have checksums in the communication protocol, communication errors are being detected, but in kncasic driver we decided to process errored messages as well, just in hope that some of the reported nonces are still valid and we want to deliver valid nonces to the pool as soon as possible. So, in some sense filtering of duplicates is a tall we pay for ASAP delivery of golden nonce.